mm2src/coins/nft.rs

```diff
@@ -753,3 +804,59 @@ where
     }
     Ok(())
 }
+
+fn check_and_redact_if_spam(value: &mut Option<String>) -> bool {
+    match value {
+        Some(s) if s.contains("http://") || s.contains("https://") => {
+            *value = Some("URL redacted for user protection".to_string());
+            true
+        },
+        _ => false,
+    }
+}
+
+fn protect_from_history_spam(tx: &mut NftTransferHistory) {
+    let collection_name_spam = check_and_redact_if_spam(&mut tx.collection_name);
+    let token_name_spam = check_and_redact_if_spam(&mut tx.token_name);
+
+    if collection_name_spam || token_name_spam {
+        tx.possible_spam = Some(true);
+    }
+}
+
+fn protect_from_nft_spam(nft: &mut Nft) -> MmResult<(), serde_json::Error> {
+    let collection_name_spam = check_and_redact_if_spam(&mut nft.collection_name);
+    let symbol_spam = check_and_redact_if_spam(&mut nft.symbol);
+    let token_name_spam = check_and_redact_if_spam(&mut nft.uri_meta.token_name);
+    let meta_spam = check_nft_metadata_for_spam(nft)?;
+
+    if collection_name_spam || symbol_spam || token_name_spam || meta_spam {
+        nft.possible_spam = Some(true);
+    }
+    Ok(())
+}
+
+fn check_nft_metadata_for_spam(nft: &mut Nft) -> MmResult<bool, serde_json::Error> {
+    if let Some(metadata_str) = &nft.metadata {
+        if let Ok(mut metadata) = serde_json::from_str::<serde_json::Map<String, Json>>(metadata_str) {
+            if check_and_redact_metadata_field(&mut metadata, "name") {
+                nft.metadata = Some(serde_json::to_string(&metadata)?);
+                return Ok(true);
+            }
+        }
+    }
+    Ok(false)
+}
+
+fn check_and_redact_metadata_field(metadata: &mut serde_json::Map<String, Json>, field: &str) -> bool {
+    match metadata.get(field).and_then(|v| v.as_str()) {
+        Some(value) if value.contains("http://") || value.contains("https://") => {
```

Comment:
here too

Reply:
renamed function to `check_spam_and_redact_metadata_field` and added doc comment.
I will add additional spam link checks later